hubploy/auth.py

```diff
@@ -182,24 +201,47 @@ def cluster_auth_gcloud(deployment, project, cluster, zone, service_key):
         'get-credentials', cluster
     ])
 
+    yield
+
 
 def cluster_auth_aws(deployment, project, cluster, zone, service_key):
     """
     Setup AWS authentication with service_key
 
     This changes *global machine state* on what current kubernetes cluster is!
     """
-    # move credentials to standard location
+    original_kubeconfig_file_loc = get_env_var_if_exists("KUBECONFIG")
+
+    # Get credentials from standard location
     service_key_path = os.path.join(
         'deployments', deployment, 'secrets', service_key
     )
-    cred_dir = os.path.expanduser('~/.aws')
-    if not os.path.isdir(cred_dir):
-        os.mkdir(cred_dir)
-    shutil.copyfile(service_key_path, os.path.join(cred_dir, 'credentials'))
 
-    subprocess.check_call(['aws', 'eks', 'update-kubeconfig',
-                           '--name', cluster, '--region', zone])
+    # Temporarily kubeconfig file
+    temp_kube_file = tempfile.NamedTemporaryFile()
+
+    original_credential_file_loc = get_env_var_if_exists("AWS_SHARED_CREDENTIALS_FILE")
+
+    try:
+
+        # Set env variable for credential file location
+        os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
+        os.environ["KUBECONFIG"] = temp_kube_file.name
+
+        subprocess.check_call(['aws', 'eks', 'update-kubeconfig',
+                               '--name', cluster, '--region', zone])
+
+        yield
+
+    finally:
+        # Unset env variable for credential file location
+        del os.environ["AWS_SHARED_CREDENTIALS_FILE"]
+        os.environ["AWS_SHARED_CREDENTIALS_FILE"] = original_credential_file_loc
```

Comment:
If the env variable was unset originally, we should leave it unset - rather than setting it to "". Apps often change behavior on wether the variable is set or not, regardless of the value.